frontend/src/entities/jobs/hpo/create/training-definitions/training-job-definition.tsx

```diff
@@ -248,7 +247,7 @@ export function TrainingJobDefinition (props: TrainingJobDefinitionProps) {
                                         // - OR the field has a value
                                         // - OR this is the max field and the min field has a value
                                         // This evaluation avoids the issue where a blank value is evaluated as a 0 on a forced safeParse even for optional fields
-                                        if ((!hyperParameter.zValidator?.isOptional || value || (index === 1 && parameterRange.MinValue)) && !nanError ) {
+                                        if ((hyperParameter.required || value || (index === 1 && parameterRange.MinValue)) && !nanError ) {
```

Comment:
I have a couple issues with what we're doing here. I don't like that you're breaking some validation out-of-band from Zod when the concept of optional exists within Zod. Shouldn't we be running the validator if it is present no matter what? If the value can be empty then our validator should be handling that.

I also don't think how it was working (by looking a property of the validator instead of just executing the validator) was the best way to go about it either for the reasons you've listed.

Reply:
I think I see what you mean; we should just confirm that it has a `zValidator` property but then in the below lines where we actually invoke the validator we should let that confirm that it's valid. 

I'll try that and see if it works like I think it should